src/hotspot/src/share/vm/utilities/globalDefinitions_gcc.hpp

```diff
@@ -235,7 +235,7 @@ inline int g_isnan(double f) { return isnand(f); }
 #elif defined(__APPLE__)
 inline int g_isnan(double f) { return isnan(f); }
 #elif defined(LINUX) || defined(_ALLBSD_SOURCE)
-inline int g_isnan(float  f) { return isnanf(f); }
+inline int g_isnan(float  f) { return isnan(f); }
```

Comment:
JDK-8178689 is still open.  here is the discussion. 
https://mail.openjdk.java.net/pipermail/portola-dev/2017-April/000077.html

This change only limits to linux/gcc.  if glibc also implement isnan for both float/double as Bob said, I am fine with this change.